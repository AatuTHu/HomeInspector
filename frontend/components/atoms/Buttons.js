import { Text, TouchableOpacity } from 'react-native'
import { buttons } from '../../styles/buttonStyles'
import React from 'react'

export default function Buttons({text,textStyle, type, onPress}) {
    
    const getButtonStyle = () => { 
     
      const styleMap = {
        start: buttons.startButton,
        save: buttons.saveButton,
        delete: buttons.deleteButton,
        nav: buttons.navButtons,
        update: buttons.updateButton,
        refresh: buttons.refreshButton,
        pin: buttons.pinButton
      };
    
      return styleMap[type] || buttons.defaultStyle;
    }

    const getTextStyle = () => {
      const textMap = {
        white : buttons.whiteButtonText,
        black : buttons.blackButtonText
      }

      return textMap[textStyle]
    }

    return (
      <TouchableOpacity style={getButtonStyle()} onPress={onPress}>
        <Text style={getTextStyle()}>{text}</Text>
      </TouchableOpacity>
    )
  
}
