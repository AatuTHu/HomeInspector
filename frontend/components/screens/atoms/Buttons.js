import { Text, TouchableOpacity } from 'react-native'
import { buttons } from '../../../styles/buttonStyles'
import React from 'react'

export default function Buttons({text, type, onPress}) {
    
    const getButtonStyle = () => { 
     
      const styleMap = {
        start: buttons.startButton,
        save: buttons.saveButton,
        delete: buttons.deleteButton,
        nav: buttons.navButtons,
        update: buttons.updateButton,
        nav: buttons.navButtons,
        refresh: buttons.refreshButton
      };
    
      return styleMap[type] || buttons.defaultStyle;
    }

    return (
      <TouchableOpacity style={getButtonStyle()} onPress={onPress}>
        <Text style={buttons.buttonText}>{text}</Text>
      </TouchableOpacity>
    )

    
}
