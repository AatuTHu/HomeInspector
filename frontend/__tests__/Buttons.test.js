import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import Buttons from '../components/screens/atoms/Buttons';

describe('Buttons', () => {

    it('Renders correctly', () => {
        const screen = render(<Buttons text = "save" type = "save"/>);
        expect(screen).toMatchSnapshot();
    })

    it('fires a function on press', () => {
        mockOnPressFunction = jest.fn();
        const  {getByText}  = render(<Buttons onPress = { mockOnPressFunction } text = "test"/>)
        const pressButton = getByText('test')
        fireEvent.press(pressButton)

        expect(mockOnPressFunction).toHaveBeenCalled();
    })

    it('fires a function one time', () => {
        mockOnPressFunction = jest.fn();
        const  {getByText}  = render(<Buttons onPress = { mockOnPressFunction } text = "test"/>)
        const pressButton = getByText('test')
        fireEvent.press(pressButton)

        expect(mockOnPressFunction).toHaveBeenCalledTimes(1);
    })

})

