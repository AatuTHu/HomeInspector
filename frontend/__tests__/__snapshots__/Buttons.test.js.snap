// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons Renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "green",
      "borderRadius": 20,
      "borderWidth": 1,
      "marginTop": 20,
      "opacity": 1,
      "padding": 10,
      "width": "95%",
    }
  }
>
  <Text
    style={
      {
        "fontSize": 20,
        "textAlign": "center",
      }
    }
  >
    save
  </Text>
</View>
`;
